### PR TITLE
🔐 `SubmitClaims` on dataverse

### PR DESCRIPTION
**Pull request by bdeneux**
### 🔐 Send `SubmitClaims` Transaction

This PR adds a new feature allowing the sending of the [SubmitClaims](https://docs.axone.xyz/contracts/axone-dataverse#executemsgsubmitclaims) execute message on the chain. It takes a signed credential, converts it to n-quad RDF, and encodes it in `base64` to be sent to the dataverse smart contract.

### 💳 Transaction Service

To achieve this, a new client has been added to handle sending transactions on the Cosmos blockchain: `tx.Client`. This client includes a transaction builder to simplify the preparation of sending a transaction.

While there is room for improvement to make transaction sending easier and more powerful, the current implementation provides a good starting point. Potential improvements include:
- More tests for error cases, such as failed transactions or signing issues.
- Building the `TxConfig` in a more appropriate place.
- Allowing the choice of denom for the keyring key and signature management.
- Typed execute message send to the smart contract 
- And many other enhancements...

### 🔑 Keyring

This PR also improves key and mnemonic management by using an interface for the `Keys` provided to the proxy or the credential generator. This change facilitates usage in a mocked environment and lays the groundwork for future improvements, such as supporting alternatives to mnemonics.

### CODE REVIEW

File: tx/client.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		return 0, 0, err
+	}
+
+	var account authtypes.BaseAccount
+	if err := account.Unmarshal(resp.GetAccount().Value); err != nil {
+		return 0, 0, err
+	}
+
+	return account.AccountNumber, account.Sequence, nil
+}
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply tx/client.go:70]: @coderabbitai thanks, done ✅

File: keys/mnemonic.go [hunk 2]
```diff
@@ -52,11 +56,23 @@
 }
 
 func (k *Key) Alg() string {
-	return "unknown"
+	return "secp256k1"
+}
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply keys/mnemonic.go:60]: https://github.com/coderabbitai thanks, done ✅

**Review by amimart — APPROVED** [review]
Wonderful! Let's go with this and improve it depending on our needs, this design already cover most of them I guess :)

**Review by ccamel — APPROVED** [review]
That's good! Thanks 👍

**Pull request merged**